core/primitives/src/transaction.rs

```diff
@@ -214,6 +214,13 @@ impl Default for ExecutionStatus {
     }
 }
 
+#[derive(BorshSerialize, BorshDeserialize, PartialEq, Clone, Default)]
+struct PartialExecutionOutcome {
```

Comment:
So this structure only exists so that we can nicely serialize and generate hashes for merklization. On one hand, it is not clear whether duplicating `status`, `receipt_ids`, and `gas_burnt` over two structures is a good idea. But on the other hand refactoring `ExecutionOutcome` just to make merkelization easier might not be a good idea either.

Let's keep it this way.